src/resolve/adapters/digitalExperienceSourceAdapter.ts

```diff
@@ -177,3 +269,28 @@ export class DigitalExperienceSourceAdapter extends BundleSourceAdapter {
 const calculateNameFromPath = (contentPath: string): string => `${parentName(contentPath)}/${baseName(contentPath)}`;
 const digitalExperienceStructure = join('BaseType', 'SpaceApiName', 'ContentType', 'ContentApiName');
 const contentParts = digitalExperienceStructure.split(sep);
+
+/**
+ * Checks if the given path belongs to the web_app base type.
+ * web_app base type has a simpler structure without ContentType folders.
+ * Structure: digitalExperiences/web_app/spaceApiName/...files...
+ */
+const isWebAppBaseType = (path: string): boolean => {
+  const pathParts = path.split(sep);
+  const digitalExperiencesIndex = pathParts.indexOf('digitalExperiences');
+  // Check if the base type (folder after digitalExperiences) is WEB_APP_BASE_TYPE
+  return (
```

Comment:
NIT: this could simplified a little, like 
```typescript
const isWebAppBaseType = (path: string): boolean => {
  const pathParts = path.split(sep);
  const digitalExperiencesIndex = pathParts.indexOf('digitalExperiences');
  return pathParts[digitalExperiencesIndex + 1] === WEB_APP_BASE_TYPE;
};
```